src/cockpit/channels/http.py

```diff
@@ -20,145 +20,140 @@
 import logging
 import socket
 import ssl
-import threading
 
-from ..channel import Channel
+from ..channel import AsyncChannel, ChannelError
+from ..jsonutil import JsonObject, get_dict, get_int, get_object, get_str, typechecked
 
 logger = logging.getLogger(__name__)
 
 
-class HttpChannel(Channel):
+class HttpChannel(AsyncChannel):
     payload = 'http-stream2'
 
-    def create_connection(self):
-        opt_address = self.options.get('address') or 'localhost'
-        opt_port = self.options.get('port')
-        opt_unix = self.options.get('unix')
-        opt_tls = self.options.get('tls')
-        logger.debug('connecting to %s:%s; tls: %s', opt_address, opt_port or opt_unix, opt_tls)
+    @staticmethod
+    def get_headers(response: http.client.HTTPResponse, binary: 'str | None') -> JsonObject:
+        # Never send these headers
+        remove = {'Connection', 'Transfer-Encoding'}
+
+        if binary != 'raw':
```

Comment:
Hmm, while we are at the subject of making things compatible with the C bridge: that never does any string comparison to the "binary" field, in particular not `"raw"`. It just checks its presence:
```c
self->binary = json_object_has_member (options, "binary");
```
The documentation says "set to true", i.e. considering it a boolean field. I have a very strong déjà vu (or rather, déjà débattu), but I don't remember the outcome any more.

Reply:
No need to debate:

docs: https://github.com/cockpit-project/cockpit/blob/main/doc/protocol.md#command-open
> * "binary": If present set to "raw"

cockpit.js: https://github.com/cockpit-project/cockpit/blob/main/pkg/lib/cockpit.js#L849
```js
        if (binary)
            command.binary = "raw";
```

I'm happy enough to change it to a presence check if you'd prefer.

Reply:
Ah, that's what I was missing. So the C bridge is just too sloppy about this then. TBH I don't even know why we have to drop Content-Length for raw streams, so I don't have a preference.

Reply:
We have to drop it for non-raw streams because Content-Length is binary data, but for text-based streams, we don't know in advance how many characters we'll pull out of the utf-8 data.